src/app/utils/query-url-sanitization.ts

```diff
@@ -54,6 +50,15 @@ function sanitizeQueryParameters(queryString: string): string {
    return result;
 }
 
+/*
+* Special characters present in the url show that the
+* segment contains an identifier
+*/
+function hasSpecialCharacters(segment: string): boolean {
+  const specialCharacters = ['.', '=', '@', '-'];
+  return specialCharacters.some((character) =>  segment.includes(character) || !!segment.match(alphaNumericRegex));
```

Comment:
`!!segment.match(alphaNumericRegex)` doesn't depend on `character` so this line can be more efficient if replaced by:
```suggestion
  return !!segment.match(alphaNumericRegex) || specialCharacters.some((character) =>  segment.includes(character));
```

That said, I am confused about the logic due to the definition of alphaNumericRegex. Maybe a few tests targeting this function directly can help document the expected behavior here.

Reply:
I ended up changing to a function. I could not get the regex to check whether the string contains both letters and numbers.